pep-0592.rst

```diff
@@ -0,0 +1,280 @@
+(e.g., using ``isinstance``).
+
+**Unknown annotations:** When a tool or a library does not support
+annotations or encounters an unknown annotation it should just ignore it
+and treat annotated type as the underlying type. For example, if we were
+to add an annotation that is not an instance of ``struct.ctype`` to the
+annotation for name (e.g.,
+``Annotated[str, 'foo', struct.ctype("<10s")]``), the unpack method
+should ignore it.
+
+**Namespacing annotations:** We do not need namespaces for annotations
+since the class used by the annotations acts as a namespace.
+
+**Multiple annotations:** It’s up to the tool consuming the annotations
+to decide whether the client is allowed to have several annotations on
+one type and how to merge those annotations.
+
+Since the ``Annotated`` type allows you to put several annotations of
+the same (or different) type(s) on any node, the tools or libraries
+consuming those annotations are in charge of dealing with potential
+duplicates. For example, if you are doing value range analysis you might
+allow this::
+
+    T1 = Annotated[int, ValueRange(-10, 5)]
+    T2 = Annotated[T1, ValueRange(-20, 3)]
+
+Flattening nested annotations, this translates to::
+
+    ``T2 = Annotated[int, ValueRange(-10, 5), ValueRange(-20, 3)]``
+
+An application consuming this type might choose to reduce these
+annotations via an intersection of the ranges, in which case ``T2``
+would be treated equivalently to ``Annotated[int, ValueRange(-10, 3)]``.
+
+An alternative application might reduce these via a union, in which case
+``T2`` would be treated equivalently to
+``Annotated[int, ValueRange(-20, 5)]``.
+
+Consumers of specific annotations may decide to not support multiple
+annotations and throw an exception (e.g. in the example above the value
+range could throw on ``T2``).
+
+Interaction with ``get_type_hints``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``get_type_hints`` will take a new argument ``include_extras`` that
+defaults to ``False`` to preserve backward compatibility. When
+``include_extras`` is ``False``, the extra annotations will be stripped
+out of the returned value. Otherwise, the annotations will be returned
+unchanged::
+
+    @struct.packedclass Student(NamedTuple):
+      name: Annotated[str, struct.ctype("<10s")]
+
+::
+
+    get_type_hints(Student) == {'name': str}
+    get_type_hints(Student, include_extras=False) == {'name': str}
+    get_type_hints(Student, include_extras=True) == {'name': Annotated[str, struct.ctype("<10s")]}
+
+Aliases & Concerns over verbosity
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Writing ``typing.Annotated`` everywhere can be quite verbose;
+fortunately, the ability to alias annotations means that in practice we
+don’t expect clients to have to write lots of boilerplate code::
+
+    T = TypeVar('T')
+    Const = Annotated[T, my_annotations.CONST]
+
+    Class C:
+      def const_method(self: Const[T]) -> int:
+        ...
+
```

Comment:
I would add a short section about rejected ideas (even if there is currently just few of them, since more can appear during the discussion).